OpenSim/Simulation/Model/GeometryPath.cpp

```diff
@@ -1278,10 +1269,10 @@ calcLengthAfterPathComputation(const SimTK::State& s,
 double GeometryPath::
 computeMomentArm(const SimTK::State& s, const Coordinate& aCoord) const
 {
-    if(!_maSolver)
-        _maSolver = new MomentArmSolver(*_model);
+    if (!_maSolver)
+        const_cast<Self*>(this)->_maSolver.reset(new MomentArmSolver(*_model));
```

Comment:
I agree about modifying member variables in const methods; this is probably one place that makes the models non-threadsafe.

1. I tried moving the allocation of this object to `extendConnectToModel()`, but then more than half of the test cases failed. The `MomentArmSolver` was trying to do something with the `Model`, but the `Model` had not yet had `initSystem()` called on it. I decided to stay focused on the task at hand.
2. I prefer using `const_cast` instead of making a member variable mutable. Mutable member variables can be changed by any const method, whereas `const_cast` localizes the mutability of the member.